torch/linalg/__init__.py

```diff
@@ -13,3 +15,248 @@
+    r"""Returns the matrix norm or vector norm of a given tensor.
+
+    This function can calculate one of eight different types of matrix norms, or one
+    of an infinite number of vector norms, depending on both the number of reduction
+    dimensions and the value of the `ord` parameter.
+
+    Args:
+        input (Tensor): The input tensor. If dim is None, x must be 1-D or 2-D, unless :attr:`ord`
+            is None. If both :attr:`dim` and :attr:`ord` are None, the 2-norm of the input flattened to 1-D
+            will be returned.
+
+        ord (int, float, inf, -inf, 'fro', 'nuc', optional): The order of norm.
+            inf refers to :attr:`float('inf')`, numpy's :attr:`inf` object, or any equivalent object.
+            The following norms can be calculated:
+
+            =====  ============================  ==========================
+            ord    norm for matrices             norm for vectors 
+            =====  ============================  ==========================
+            None   Frobenius norm                2-norm
+            'fro'  Frobenius norm                -- not supported --
+            'nuc'  nuclear norm                  -- not supported --
+            inf    max(sum(abs(x), dim=1))       max(abs(x)) 
+            -inf   min(sum(abs(x), dim=1))       min(abs(x)) 
+            0      -- not supported --           sum(x != 0) 
+            1      max(sum(abs(x), dim=0))       as below 
+            -1     min(sum(abs(x), dim=0))       as below
+            2      2-norm (largest sing. value)  as below 
+            -2     smallest singular value       as below
+            other  -- not supported --           sum(abs(x)**ord)**(1./ord)
+            =====  ============================  ==========================
+
+            Default: ``None``
+
+        dim (int, 2-tuple of ints, 2-list of ints, optional): If :attr:`dim` is an int,
+            vector norm will be calculated over the specified dimension. If :attr:`dim`
+            is a 2-tuple of ints, matrix norm will be calculated over the specified
+            dimensions. If :attr:`dim` is None, matrix norm will be calculated
+            when the input tensor has two dimensions, and vector norm will be
+            calculated when the input tensor has one dimension. Default: ``None``
+
+        keepdim (bool, optional): If set to True, the reduced dimensions are retained
+            in the result as dimensions with size one. Default: ``False``
+
+        out (Tensor, optional): The output tensor. Ignored if ``None``. Default: ``None``
+
+        dtype (:class:`torch.dtype`, optional): The desired data type of
+            returned tensor. If specified, the input tensor is casted to
+            :attr:`dtype` while performing the operation. This functionality is not
+            currently supported for :attr:`ord='nuc'` or :attr:`ord='fro'`. Default: ``None``
+
+    Examples::
```

Comment:
I copied all these examples from `numpy.linalg.norm`. `torch.functional.norm` had a handful of them, but not all. Let me know if this is overkill and I can remove some.

Reply:
Do we really need to bind this function like this, as opposed to the way outer is bound (as of this base, it's actually being fixed to be torch.outer)?